client/Profile.jsx

```diff
@@ -0,0 +1,195 @@
+            { name: 'challengeBegin', label: 'Challenge phase begins' },
+            { name: 'attackersDeclared', label: 'Attackers declared' },
+            { name: 'defendersDeclared', label: 'Defenders declared' },
+            { name: 'winnerDetermined', label: 'Winner determined' },
+            { name: 'dominance', label: 'Dominance phase' },
+            { name: 'standing', label: 'Standing phase' }
+        ];
+    }
+
+    componentWillReceiveProps(props) {
+        this.setState({ email: props.user.email, disableGravatar: props.user.settings.disableGravatar || false, promptedActionWindows: props.user.promptedActionWindows || this.windowDefaults });
+    }
+
+    onChange(field, event) {
+        var newState = {};
+
+        newState[field] = event.target.value;
+        this.setState(newState);
+    }
+
+    onWindowToggle(field, event) {
+        var newState = {};
+        newState.promptedActionWindows = this.state.promptedActionWindows;
+
+        newState.promptedActionWindows[field] = event.target.checked;
+        this.setState(newState);
+    }
+
+    onSaveClick(event) {
+        event.preventDefault();
+
+        this.setState({ errorMessage: undefined, successMessage: undefined });
+
+        this.verifyEmail();
+        this.verifyPassword(true);
+        
+        if(_.any(this.state.validation, function(message) {
+            return message && message !== '';
+        })) {
+            this.setState({ errorMessage: 'There was an error in one or more fields, please see below, correct the error and try again' });
+            return;
+        }
+
+        this.setState({ loading: true });
+
+        $.ajax('/api/account/' + this.props.user.username, 
+            { 
+                method: 'PUT',
+                data: { data: JSON.stringify({
+                    email: this.state.email,
+                    password: this.state.newPassword,
+                    promptedActionWindows: this.state.promptedActionWindows,
+                    settings : {
+                        disableGravatar: this.state.disableGravatar
+                    }
+                }) }
+            })
+            .done((data) => {
+                if(data.success) {
+                    this.setState({ successMessage: 'Profile saved successfully' });
+
+                    this.props.socket.emit('authenticate', data.token);
+                    this.props.refreshUser(data.user, data.token);
+                } else {
+                    this.setState({ errorMessage: data.message });
+                }
+            })
+            .always(() => {
+                this.setState({ loading: false });
+            });
+    }
+
+    verifyPassword(isSubmitting) {
+        var validation = this.state.validation;
+
+        delete validation['password'];
+
+        if(!this.state.newPassword && !this.state.newPasswordAgain) {
+            return;
+        }
+
+        if(this.state.newPassword.length < 6) {
```

Comment:
Either now or in a separate PR we should consider consolidating these rules (along with email validation) into shared logic so that we remain consistent. This will be the third place we validate password on the client side.